src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11KeyPairGenerator.java

```diff
@@ -99,29 +99,35 @@
         // set default key sizes and apply our own algorithm-specific limits
         // override lower limit to disallow unsecure keys being generated
         // override upper limit to deter DOS attack
```

Comment:
No a P11 expert, but I assume `algorithm` here is already guaranteed to be in uppercase?

Reply:
Yes, for P11KeyPairGenerator, its algorithm values are all in uppercase. I verified it with an existing regression test.